bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```
@@ -0,0 +1,290 @@
+            actualPowerValue = ((1 - interpolation) * powerFloor) + (interpolation * powerCeiling);
+            return actualPowerValue / 1000.0;
+        } // else both null - this shall not happen
+        return UNDEF;
+    }
+
+    public double getDayTotal(LocalDate queryDate) {
+        if (rawData.isEmpty()) {
+            return UNDEF;
+        }
+        JSONObject contentJson = new JSONObject(rawData.get());
+        JSONObject resultJson = contentJson.getJSONObject("result");
+        JSONObject wattsDay = resultJson.getJSONObject("watt_hours_day");
+
+        if (wattsDay.has(queryDate.toString())) {
+            return wattsDay.getDouble(queryDate.toString()) / 1000.0;
+        }
+        return UNDEF;
+    }
+
+    public double getRemainingProduction(ZonedDateTime queryDateTime) {
+        if (wattHourMap.isEmpty()) {
+            return UNDEF;
+        }
+        double daily = getDayTotal(queryDateTime.toLocalDate());
+        double actual = getActualValue(queryDateTime);
+        if (daily < 0 || actual < 0) {
+            return UNDEF;
+        }
+        return daily - actual;
+    }
+
+    public ZoneId getZone() {
+        return zone;
+    }
+
+    @Override
+    public String toString() {
+        return "Expiration: " + expirationDateTime + ", Valid: " + valid + ", Data:" + wattHourMap;
+    }
+
+    /**
+     * SolarForecast Interface
+     */
+    @Override
+    public State getDay(LocalDate localDate, String... args) {
+        if (args.length > 0) {
+            logger.info("ForecastSolar doesn't accept arguments");
+            return UnDefType.UNDEF;
+        }
+        double measure = getDayTotal(localDate);
+        return Utils.getEnergyState(measure);
+    }
+
+    @Override
+    public State getEnergy(Instant start, Instant end, String... args) {
+        if (args.length > 0) {
+            logger.info("ForecastSolar doesn't accept arguments");
+            return UnDefType.UNDEF;
```

Is this expected to happen? Perhaps you could instead throw `IllegalArgumentException` to fail fast?